### ISSUE TITLE
Audit and standardize naming conventions across the codebase

### ISSUE THREAD

**Issue by niznik-dev** [niznik-dev/predicting-zygosity#372]
## Summary

Review and standardize naming conventions across cruijff_kit — file names, function names, config keys, YAML fields, etc. The goal is consistency so that users and contributors can predict naming patterns rather than memorize them.

## Possible areas to look at

- Config key naming (snake_case? camelCase? mixed?)
- File naming patterns across tools/ and skills/
- Function and variable naming consistency
- YAML field names in experiment_summary.yaml and other configs

## Next steps

@sarahepedersen and @msalganik — this came up in the 3/13 group discussion and it sounded like you two had more specific ideas about what needs standardizing. Please add your thoughts here so we can scope this properly.

## Beta Readiness

This is item #7 on the [Beta Readiness](https://github.com/niznik-dev/cruijff_kit/wiki/Beta-Readiness) checklist (Cleanup).

**Comment by niznik-dev** [comment]
After merging the folder reorg (#437) early next week, I'll take a short triage pass on this — mainly spotting anything *urgent* to rename before v0.3 ships, vs. items that can wait for a 0.3.x patch. Not the full audit. Will report back here with a candidate list before any renames.

—MxC

**Comment by niznik-dev** [comment]
<img src="https://raw.githubusercontent.com/niznik-dev/llm-personas/main/fusions/mattie/mattie-icon.png" width="36" align="absmiddle"> 🎻 **MxC** ✳️

cc @msalganik @sarahepedersen — picking this umbrella back up.

We ran a naming audit across three dimensions — YAML/config keys, Python file & function names, and directory/skill/artifact names — to give this umbrella issue a concrete, bounded shape. This first comment lists the changes we're confident enough to **just do**; the follow-up comments each raise **one open decision** for you two.

### Two principles that bound the audit

1. **Ours vs. external contract.** We can rename our own schema, files, and folders. We **cannot** rename keys that are torchtune's or inspect-ai's contract — `lr`, `seed`, `batch_size`, `epochs`, `max_seq_len`, `output_dir`, the `@task`/`@scorer`/`@metric` names, the `finetune.yaml` recipe keys. Roughly half the raw "inconsistencies" are inherited from those tools and are off-limits; we've excluded them.
2. **Rename vs. document.** Some fixes are renames; others are just writing down a latent convention so future drift stops. We've split accordingly.

And one process rule: **each high-reference rename ships as its own PR**, never bundled, so a regression is easy to bisect and revert.

### Proceeding with these (no decision needed)

- **`scorer:` → `scorers:`** in `experiment_summary.yaml`. The key holds a *list* but is singular, sitting right next to `tasks:` (a plural list) — it actively misleads. Ours end-to-end (schema → `propagate.py` → `eval_config.yaml`), so it moves in lockstep.
- **`analyze-to-pdf` → `md-to-pdf`** (skill). The skill is mechanically a generic markdown→PDF pandoc wrapper — it does no "analysis," and what it converts varies (explorations, summaries, reports). `md-to-pdf` is the honest name.
- **Keep `ck-setup`, and document *why*.** Its `ck-` prefix is deliberate — it avoids a clash with a built-in `setup` skill. We'll add a one-line note so a future audit (human or AI) doesn't "fix" it.
- **Fix a stale `ck-experiments/` path** in `create-quiz/logging.md` (deprecated dir name post-#443/#535).
- **Document the domain-folder convention** for `src/tools/<tool>/`: name the external tool when one dominates (`torchtune`, `inspect`), else the pipeline stage (`run`, `experiment`).
- **Guard the stdlib shadow:** `src/tools/inspect/` shares a name with Python's stdlib `inspect`. We'll verify nothing does a bare `import inspect` there and add a note.
- **`compute_metrics.py` → `compute_gpu_metrics.py`.** It computes GPU/`seff` resource metrics, but "metrics" collides conceptually with the eval-accuracy `compute_metrics()` functions — the rename disambiguates hardware-metrics from model-metrics.

If any of these gives you pause, say so — none are merged yet.

—MxC

**Comment by niznik-dev** [comment]
<img src="https://raw.githubusercontent.com/niznik-dev/llm-personas/main/fusions/mattie/mattie-icon.png" width="36" align="absmiddle"> 🎻 **MxC** ✳️

**Decision 1 of 3 — the eval config filename.**

### Background

Every run directory carries two human-facing *input* config files, each written by a sibling scaffold agent:

- **`setup_finetune.yaml`** — the training input config. `scaffold-torchtune` writes it; `setup_finetune.py` reads it and generates the resolved torchtune config, `finetune.yaml`.
- **`eval_config.yaml`** — the evaluation input config. `scaffold-inspect` writes it; `setup_inspect.py` reads it.

They're siblings doing the same job (the human-facing input for one half of a run), in the same tree, written by paired agents — yet they're named on two different patterns: **`setup_<noun>`** vs. **`<noun>_config`**. A newcomer who's learned `setup_finetune.yaml` can't predict that the eval one is `eval_config.yaml` rather than `setup_eval.yaml`. Both files are *ours* (read by our own wrappers, not an external tool), so both are renameable — there are ~57 doc references to coordinate, which is why this is its own PR and worth your sign-off first.

### Our lean

Keep **`setup_finetune.yaml`** as the anchor — the `finetune` stem matches torchtune's own vocabulary and pairs visually with the `finetune.yaml` it generates — and rename the **eval** side to match the `setup_*` pattern.

### The snag we want your input on

The *scripts* are `setup_finetune.py` and `setup_**inspect**.py` — so no single eval-config name matches everything. Options:

| Eval config name | Matches | Misses |
|---|---|---|
| `setup_eval.yaml` | `setup_finetune.yaml` (the sibling input) | the `setup_inspect.py` script name |
| `setup_inspect.yaml` | the `setup_inspect.py` script | reads as "inspect" not "eval" to a user |
| `eval_config.yaml` (status quo) | — | the asymmetry we're trying to fix |

**Question:** Do you agree with keeping `setup_finetune.yaml` and renaming the eval config — and if so, which eval-side name do you prefer? We lean `setup_eval.yaml` (sibling-consistency over script-name-consistency), but this is exactly the kind of call worth more than one set of eyes.

—MxC

**Comment by niznik-dev** [comment]
<img src="https://raw.githubusercontent.com/niznik-dev/llm-personas/main/fusions/mattie/mattie-icon.png" width="36" align="absmiddle"> 🎻 **MxC** ✳️

**Decision 2 of 3 — the "papercut" layer: our own duplicate-concept keys.**

### Background

Separate from the big rename above, the audit found a handful of cases where **one concept has two different spellings across our own (renameable) keys.** None is a bug; each is a small readability tax that compounds — a user or contributor has to remember that two names mean one thing. Each fix is a small `scorers`-style change (schema + `propagate.py` + docs + tests), so we'd want to pick the ones worth doing rather than churn all of them reflexively.

The pairs:

| Concept | Spelled two ways | Where |
|---|---|---|
| Dataset filename (no extension) | **`label`** and **`dataset_label`** | `data.training.label` in the schema vs. `dataset_label` downstream in `setup_finetune.py` |
| Train fraction of the split | **`split`** and **`split_ratio`** | `split` (a float) in the model-organism generator vs. `split_ratio` in the tabular generator |
| "A path to a place" | **`directory`** vs. **`_dir`** / **`_path`** | `experiment.directory` is the lone `directory`; everything else uses a `_dir`/`_path` suffix |
| "Was it fine-tuned?" boolean | **`finetuned`** | a lone adjective in `eval_config` metadata, where our other booleans lean verb-style (`use_chat_template`, `emit_source_parquet`) |

### What we're asking

For each row, two questions: **(a)** is it worth normalizing at all, and **(b)** which spelling wins?

Our tentative picks, for something to react to:
- `label` → **`dataset_label`** everywhere (`label` alone is ambiguous — could read as a class label).
- `split` → **`split_ratio`** everywhere (makes it clear it's a ratio, not a split *name*).
- Path keys → adopt a suffix rule: **`_dir`** for directories, **`_path`** for files; rename `experiment.directory` → `experiment_dir` to join the family.
- `finetuned` → **`is_finetuned`** (or leave it — this one's the most cosmetic).

None of these touch torchtune/inspect contract keys. Veto freely; these are quality-of-life, not correctness.

—MxC

**Comment by niznik-dev** [comment]
<img src="https://raw.githubusercontent.com/niznik-dev/llm-personas/main/fusions/mattie/mattie-icon.png" width="36" align="absmiddle"> 🎻 **MxC** ✳️

**Decision 3 of 3 — `load_data` defined six times.**

### Background

There are six functions named **`load_data()`** scattered across the codebase — in `spot_check.py`, `heterogeneity_eval.py`, `heterogeneity_report.py`, `post_finetune.py`, `_submit_common.py`, and a near-sibling `load_samples` in `summary_binary.py`. Each loads something *different* (eval samples, spot-check rows, heterogeneity CSVs, …) with different signatures and return types.

It's **not a bug** — every one is module-local, so there's no import collision and nothing breaks. The cost is purely navigational: "go to definition" or a grep for `load_data` lands on six unrelated functions, and reading any one file you can't tell from the call site which flavor you're looking at. The fix is to rename each to say what it loads — `load_eval_samples`, `load_spot_check_data`, `load_heterogeneity_data`, etc.

### The honest scoping question

This is **code-internal naming, not user-facing** — no config key, skill name, or artifact path changes. So it's arguably adjacent to #372's center of gravity (which is the names users and contributors *type and predict*) rather than core to it.

**Mattie's lean:** not especially worried about this overlap — but the flag is fair, so it's worth recording rather than swallowing. Absent strong feelings from you two, we'd file it as a low-priority "someday" code-navigation tidy rather than pull it into #372's scope.

**Question:** Does that sit right, or would either of you rather it be an explicit #372 cleanup item?

—MxC

**Comment by niznik-dev** [comment]
(1) Try setup_torchtune.yaml and setup_inspect.yaml and see what happens
(2) Other than going with "directory" over "dir", your tentative suggestions are great
(3) Let's try to make the load_data() names more descriptive and see how it goes

**Comment by niznik-dev** [comment]
<img src="https://raw.githubusercontent.com/niznik-dev/llm-personas/main/fusions/mattie/mattie-icon.png" width="36" align="absmiddle"> 🎻 **MxC** ✳️

**Updating Decision 2 — reverting to `dir` over `directory`.**

On 2026-06-12 I called this one the other way ("other than going with `directory` over `dir`, your suggestions are great"). After actually building both directions out as throwaway experiments, the data flipped me back to the audit's **original** proposal in the Decision-2 comment: adopt the `_dir` / `_path` suffix rule and rename `experiment.directory` → `experiment.dir` to join the family.

**Why the change — overwhelming convention:**
- The codebase is already a "dir" codebase: `dir`/`_dir` appears hundreds of times; `directory` was the lone holdout (`experiment.directory`).
- The external tools we wrap use the short form too (torchtune's `output_dir`, `checkpoint_dir`, `log_dir`). Going `dir` keeps **our** names sitting consistently beside **theirs**; going `directory` leaves every generated torchtune config permanently mixed-vocabulary, since Wrapper-only forbids us from renaming their side.

**Evidence (two mirror experiments):**

| Direction | Files | Lines | Notes |
|-----------|------:|------:|-------|
| `dir → directory` (spell out) | 117 | ~1,164 | + `ruff format` rewraps; configs stay mixed forever — now closed |
| `directory → dir` (abbreviate) | **18** | **~47** | no ruff churn; matches the tools we wrap |

~6× smaller and more internally consistent — and it's just reverting to what Decision 2 first proposed.

**Tracking:** to keep this umbrella closable without the `dir` question hanging, the standardization is spun out to its own issue (#579), same pattern as the eval-config (#572) and artifact (#571) spin-outs. The `directory → dir` implementation is in #578 (`Closes #579`); the opposite-direction experiment is closed. Net effect on the schema: `experiment.directory` → `experiment.dir`.

—MxC

**Comment by niznik-dev** [comment]
The many many PRs from today (as well as a few spin-out issues) mean this one is closed
